include/jwt.h

```diff
@@ -320,6 +320,183 @@ DEPRECATED(JWT_EXPORT int jwt_del_grant(jwt_t *jwt, const char *grant));
+ * try to add a header that already exists.
+ *
+ * @param jwt Pointer to a JWT object.
+ * @param header String containing the name of the header to add.
+ * @param val int containing the value to be saved for header.
+ * @return Returns 0 on success, valid errno otherwise.
+ *
+ * Note, this only allows for integer based headers. If you wish to add
+ * string headers, then use jwt_add_header(). If you wish to add more
+ * complex headers (e.g. an array), then use jwt_add_headers_json().
+ */
+JWT_EXPORT int jwt_add_header_int(jwt_t *jwt, const char *header, long val);
+
+/**
+ * Add a new boolean header to this JWT object.
+ *
+ * Creates a new header for this object. The string for header
+ * is copied internally, so do not require that the pointer or string
+ * remain valid for the lifetime of this object. It is an error if you
+ * try to add a header that already exists.
+ *
+ * @param jwt Pointer to a JWT object.
+ * @param header String containing the name of the header to add.
+ * @param val boolean containing the value to be saved for header.
+ * @return Returns 0 on success, valid errno otherwise.
+ *
+ * Note, this only allows for boolean based headers. If you wish to add
+ * string headers, then use jwt_add_header(). If you wish to add more
+ * complex headers (e.g. an array), then use jwt_add_headers_json().
+ */
+JWT_EXPORT int jwt_add_header_bool(jwt_t *jwt, const char *header, int val);
+
+/**
+ * Add headers from a JSON encoded object string.
+ *
+ * Loads a header from an existing JSON encoded object string. Overwrites
+ * existing header. If header is NULL, then the JSON encoded string is
+ * assumed to be a JSON hash of all headers being added and will be merged
+ * into the header listing.
+ *
+ * @param jwt Pointer to a JWT object.
+ * @param json String containing a JSON encoded object of headers.
+ * @return Returns 0 on success, valid errno otherwise.
+ */
+JWT_EXPORT int jwt_add_headers_json(jwt_t *jwt, const char *json);
+
+/**
+ * Delete a header from this JWT object.
+ *
+ * Deletes the named header from this object. It is not an error if there
+ * is no header matching the passed name. If header is NULL, then all headers
+ * are deleted from this JWT.
+ *
+ * @param jwt Pointer to a JWT object.
+ * @param header String containing the name of the header to delete. If this
+ *    is NULL, then all headers are deleted.
+ * @return Returns 0 on success, valid errno otherwise.
+ */
+JWT_EXPORT int jwt_del_headers(jwt_t *jwt, const char *header);
+
+/**
+ * @deprecated
+ * Delete a header from this JWT object.
+ *
+ * Deletes the named header from this object. It is not an error if there
+ * is no header matching the passed name.
+ *
+ * @param jwt Pointer to a JWT object.
+ * @param header String containing the name of the header to delete.
+ * @return Returns 0 on success, valid errno otherwise.
+ */
+DEPRECATED(JWT_EXPORT int jwt_del_header(jwt_t *jwt, const char *header));
```

Comment:
You removed the definition, but left this declaration.

Reply:
This should be fixed.